lib/std/crypto/aegis.zig

```diff
@@ -965,6 +991,31 @@ test "Aegis256X4 test vector 1" {
     try testing.expectError(error.AuthenticationFailed, Aegis256X4_256.decrypt(&empty, &empty, tag256, &empty, nonce, key));
 }
 
+test "Aegis256X failed authentication" {
+    const aegis = Aegis256;
+
+    const key = [_]u8{0x00} ** aegis.key_length;
+    const nonce = [_]u8{0x00} ** aegis.nonce_length;
```

Comment:
```suggestion
    const key: [aegis.key_length]u8 = @splat(0);
    const nonce: [aegis.nonce_length]u8 = @splat(0);
```
And the same for the other locations where the length is known prior.